docs/resources/pumls/model.puml

```diff
@@ -0,0 +1,58 @@
+@startuml Model
+
+!includeurl ./style.puml
+
+package "Model" {
+
+    class ModelMetrics {
+        - _accuracy: float
+        - _precision: float
+        - _recall: float
+        - _f1_score: float
+        - _loss: float
+        + accuracy(): float
+        + precision(): float
+        + recall(): float
+        + f1_score(): float
+        + loss(): float
+    }
+
+    class ModelDetails {
```

Comment:
Add `__init__(name: str, version: str, framework: str, target: str, dtype: str)` and `__iter__(): Iterator[Tuple[str, Any]]` functions